### PR TITLE
QA best practices (EGM-298)

### PR DESCRIPTION
**Pull request by rasika-purohit**
[skip ci]

### CODE REVIEW

File: docs/bot-builder/basic/qa-best-practices.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+---
+title: Best Practices for Quality Assessment
+---
+
+### 1. All functional flows verification (as per IVA design document):
```
Comment (sanoopnair) [inline docs/bot-builder/basic/qa-best-practices.md:5]: before the functional testing QA's are also involved in Architecture Review and we try to find flaws in the design flows. 

should we include this part ?

File: docs/bot-builder/basic/qa-best-practices.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+Testing all the functional flows will ensure if the bot is giving correct responses to the user. It will also ensure if the users are going through the journey as desgined during the Conversation Design phase.
+
+* Test of all the positive and negative flows as per the design.
+* Identify and write different scenarios before test execution.
+* Execution of scenarios, and issues fixes with retesting.
```
Comment (sanoopnair) [inline docs/bot-builder/basic/qa-best-practices.md:11]: while testing the flows QA's also create bot qa sheet for test

File: docs/bot-builder/basic/qa-best-practices.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+* Verify the API responses are as per requirements.
+* Verfify API errors.
+* Verify API response time (should be not more than 20 seconds).
+* Verification of different APIs responses handling on the IVA.
+* Verify the responses for different types of test data.
+
+### 3. Small talk verification:
+
+Small Talk is a module present for all the bot. Small Talk responds to random questions asked by user such as _Are yuo a robot?_.
```
Comment (sanoopnair) [inline docs/bot-builder/basic/qa-best-practices.md:25]: here also we should check positive and negative statements

File: docs/bot-builder/basic/qa-best-practices.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+- Functional flows should work same as it works for primary language.
+- Adding different language should not impact English language flow.
+- Bot break message language should be checked.
+
+### 10. Chat agent functionality (if enabled): 
+
+* Check if the chat is transferred to agent on bot break or upon user says.
+* Check if the chat gets assigned to an agent and if you are getting reponses on the bot.
+* Check if agent is able to close the chat.
+
```
Comment (sanoopnair) [inline docs/bot-builder/basic/qa-best-practices.md:104]: Analytics part is missing , we should check that too

Comment (rasika-purohit) [reply docs/bot-builder/basic/qa-best-practices.md:104]: @sanoopnair who can help me with the checklist for testing analytics?

File: docs/bot-builder/basic/qa-best-practices.md (hunk 1 above)
Comment (rasika-purohit) [reply docs/bot-builder/basic/qa-best-practices.md:5]: Solid point to be present on the document. We will incorporate this in the Introduction, as Best Practices will mostly be accessed towards the end of the development lifecycle.

File: docs/bot-builder/basic/qa-best-practices.md (hunk 4 above)
Comment (sanoopnair) [reply docs/bot-builder/basic/qa-best-practices.md:109]: QA's From IMPL team

**Review by medhaanand — APPROVED** [review]

**Pull request merged**